state-chain/chains/src/btc/vault_swap_encoding.rs

```diff
@@ -20,6 +23,8 @@ impl Encode for UtxoEncodedData {
 	fn encode(&self) -> Vec<u8> {
 		let mut r = Vec::with_capacity(MAX_NULLDATA_LENGTH);
 
+		r.push(CURRENT_VERSION);
```

Comment:
We should encode the version here.

```suggestion
		CURRENT_VERSION.encode_to(&mut r);
```

Reply:
What do you mean? Encoding of a single byte is just the byte itself... If this is just about style, I'm okay either way, although i did use push intentionally to make it slightly more explicit, but I also understand the consistency argument.